demo/api/src/pages/pages.resolver.ts

```diff
@@ -44,7 +44,7 @@ export class PagesResolver {
 
     @ResolveField(() => PageTreeNode, { nullable: true })
     async pageTreeNode(@Parent() page: Page): Promise<PageTreeNodeInterface | null> {
-        return this.pageTreeService.createReadApi().getFirstNodeByAttachedPageId(page.id);
+        return this.pageTreeService.createReadApi({ visibility: "all" }).getFirstNodeByAttachedPageId(page.id);
```

Comment:
We should inject PageTreeReadApiService instead, that sets visibility correctly (accodring to passed headers)

Reply:
I think `visibility: "all"` is correct here, since this is a field resolver in the PagesResolver. 

If I query a page by its ID and I specify in the GraphQL query that I want to get the pageTreeNode associated to this page, then I always want to get the pageTreeNode. I can't think of a scenario where I would want the API to not return the pageTreeNode.

Reply:
I switched this with `PageTreeReadApiService` in https://github.com/vivid-planet/comet/pull/1023/commits/7a41d1f3eb5deda00a1939e0b4304e15d148d175